appcues/src/main/java/com/appcues/RenderContextManager.kt

```diff
@@ -70,10 +72,25 @@ internal class RenderContextManager(override val scope: Scope) : KoinScopeCompon
     }
 
     fun getPotentialExperiences(renderContext: RenderContext): List<Experience> {
-        return potentiallyRenderableExperiences[renderContext] ?: arrayListOf()
+        return arrayListOf<Experience>().apply {
+            previewExperiences[renderContext]?.let { add(it) }
+            potentiallyRenderableExperiences[renderContext]?.let { addAll(it) }
+        }
     }
 
     fun stop() {
         stateMachineSlots.values.forEach { it.stop() }
     }
+
+    fun putPreview(experience: Experience) {
+        val renderContext = experience.renderContext
+
+        // check if context is embed
+        if (renderContext is Embed) {
+            // if frame of that embed is null, also remove its state machine from slot
+            stateMachineSlots.remove(renderContext)
```

Comment:
can you say more about this part? the comment makes it sound like it's checking a condition on the frame but it is not actually?

Reply:
so instead of "show" on a existing state machine with an existing experience, we need to remove the state machine and create a new every time, or else the "show" logic will try to dismiss the existing experience, and this might be a problem because the frame might not be visible, thus living the coroutine in a wait state forever.

Reply:
to reproduce this, lets say we preview frame1, the first time everything is going to work OK, but then if you navigate back to events fragment and trigger the deep link again, then nothing happens. this is because on the second time we would be trying to dismiss an experience that is already being rendered on that state machine, but dismiss is impossible because since the view is not visible, the viewModel does not reply back.

so this is one of the solutions, the other would be to enhance how we dismiss experience to make sure this doesnt happen i guess.

Reply:
the commend line doesnt make sense actually, it was part of a different logic I was trying to put in here, it should say something like

"every time we put a new preview for embed, silently remove existing state machine.

Reply:
aha ok i think i follow now thank you. Ran into similar things while reviewing the other code around error handling. Makes sense to consider a preview similar to a re-register of a frame I think, where the new preview will trigger the creation of a new state machine for that context.

I think this can adapt to work on top of the other changes proposed in https://github.com/appcues/appcues-android-sdk/pull/460 fairly easily, but will require some merge handling depending on what goes first.